### PR TITLE
Remove QMSI sensor subsystem ADC driver and use Designware driver for ARC. 

### PR DESCRIPTION
**Pull request by punitvara**
- Remove old adc_qmsi_ss driver and use DesignWare driver.
- Make DW driver to use dts.
- Update grove sensors ( light and temperature sensor) driver to use new ADC APIs.
- Update adc_api test for ARC board and it working fine. 
- Addressed all comments from #8857

### CODE REVIEW

**Comment by punitvara** [comment]
@nashif  @MaureenHelm  Your review comments are addressed given on #8857. Please let me know if you feel I should provide any updates.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for fixing all the direct references to `adc_info_dev`, that part looks good now. There are additional comments in #9602 that still need to be addressed. Please also start running sanitycheck locally and fix the failures. You can do it like this:
```
$ sanitycheck -T tests/drivers/adc -T samples/sensor/grove_light -l
```

Please run gitlint on all of your commits, there is an issue with this one:
```
Commit b538745b9a:
1: UC3 Title does not follow [subsystem]: [subject]: "boards:quark_se_c1000_ss_devboard: Enable ADC node"
```

**Comment by punitvara** [comment]
@MaureenHelm I have addressed all your comments from #9602 and #8857 
Can you please review once ?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please fix the CI failures. Instructions to run them locally are here: https://github.com/zephyrproject-rtos/zephyr/pull/9605#pullrequestreview-149307124

File: drivers/adc/adc_dw.c [hunk 1]
```diff
@@ -153,17 +168,19 @@ static void adc_goto_normal_mode(struct device *dev)
 	/*Poll waiting for normal mode*/
 	do {
 		reg_value = sys_in32(PERIPH_ADDR_BASE_CREG_SLV0);
-	} while ((reg_value & 0x1) == 0);
+	} while ((reg_value & BIT(3)) == 0);
 }
 #endif
 
+/* TODO: Enable ADC power down capability */
+#ifdef CONFIG_ADC_DW_POWER_DOWN
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/adc_dw.c]: ok. Removed.

File: drivers/adc/adc_dw.c [hunk 2]
```diff
@@ -338,154 +455,90 @@ int adc_dw_init(struct device *dev)
...
+static const struct adc_driver_api api_funcs = {
+	.channel_setup = adc_dw_channel_setup,
+	.read          = adc_dw_read,
+#ifdef CONFIG_ADC_ASYNC
+	.read_async    = adc_dw_read_async,
 #endif
-	};
+};
 
 static struct adc_config adc_config_dev = {
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/adc_dw.c]: ok. Done

File: samples/sensor/grove_temperature/prj.conf [hunk 3]
```diff
@@ -7,4 +7,5 @@ CONFIG_GROVE=y
 CONFIG_GROVE_TEMPERATURE_SENSOR_ADC_CHANNEL=10
 CONFIG_GROVE_TEMPERATURE_SENSOR=y
 CONFIG_GROVE_TEMPERATURE_SENSOR_V1_X=y
-CONFIG_GROVE_LCD_RGB=y
+
+#CONFIG_GROVE_LCD_RGB=y
```
[the remark this answers is not in the document]
Comment (punitvara) [reply samples/sensor/grove_temperature/prj.conf]: Done.

File: drivers/adc/adc_dw.c [hunk 4]
```diff
@@ -200,116 +270,164 @@ static void adc_dw_enable(struct device *dev)
...
+	if (channel_id >= DW_CHANNEL_COUNT) {
+		return -EINVAL;
+	}
 
-	sys_out32(sys_in32(adc_base + ADC_SET)|ADC_FLUSH_RX, adc_base + ADC_SET);
-	irq_unlock(saved);
+	info->active_channels |= 1 << channel_id;
 
-	info->state = ADC_STATE_DISABLED;
+	adc_dw_enable(dev);
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/adc_dw.c]: done.

File: drivers/adc/adc_dw.c [hunk 5]
```diff
@@ -200,116 +270,164 @@ static void adc_dw_enable(struct device *dev)
...
+		return -EINVAL;
+	}
+
+	/* Check the requested channels are subset of total active channels */
+	for (i = 0; i < DW_CHANNEL_COUNT; i++) {
+		if (((seq_tbl->channels >> i) & 0x1) &&
+				(BIT(i) & info->active_channels)) {
+			info->channels |= BIT(i);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/adc_dw.c]: ok

File: drivers/adc/adc_dw.c [hunk 6]
```diff
@@ -200,116 +270,164 @@ static void adc_dw_enable(struct device *dev)
...
 	struct adc_info *info = dev->driver_data;
-	const struct adc_config *config = dev->config->config_info;
-	u32_t adc_base = config->reg_base;
 
-	sys_out32(ADC_INT_DSB|ENABLE_ADC, adc_base + ADC_CTRL);
-	adc_goto_deep_power_down();
-	sys_out32(ADC_INT_DSB|ADC_SEQ_PTR_RST, adc_base + ADC_CTRL);
+	/* There is no provision in the IP to set
+	 * gain and reference voltage settings. So
+	 * ignoring these settings.
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/adc_dw.c]: ok

**Comment by punitvara** [comment]
@MaureenHelm  Thanks for review. I addressed all your comments and also tested changes with grove sensors. Everything is working fine. Please have a look once.

File: drivers/adc/Kconfig.dw [hunk 7]
```diff
@@ -77,20 +54,6 @@ config ADC_DW_FALLING_EDGE
 
 endchoice
 
-config ADC_DW_SAMPLE_WIDTH
```
[the remark this answers is not in the document]
Comment (punitvara) [reply drivers/adc/Kconfig.dw]: Done

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 8]
```diff
@@ -104,6 +104,14 @@
 #define ADC_1ST_CHANNEL_ID	0
 #define ADC_1ST_CHANNEL_INPUT	0
 
+#elif defined(CONFIG_ARC)
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply tests/drivers/adc/adc_api/src/test_adc.c]: This still needs to be fixed

File: samples/sensor/grove_temperature/src/main.c [hunk 9]
```diff
@@ -41,9 +43,12 @@ void main(void)
 #endif
 
 	while (1) {
-		struct sensor_value temp;
 
-		sensor_sample_fetch(dev);
+		read = sensor_sample_fetch(dev);
+		if (read) {
+			printk("SAMLPE\n");
```
Comment (MaureenHelm) [inline samples/sensor/grove_temperature/src/main.c]: Typo

File: tests/drivers/adc/adc_api/testcase.yaml [hunk 10]
```diff
@@ -3,23 +3,4 @@ common:
...
-    platform_whitelist: quark_se_c1000_ss_devboard
-  peripheral.adc.resolution_6:
-    extra_configs:
-      - CONFIG_ADC_QMSI_SAMPLE_WIDTH=6
-    platform_whitelist: quark_se_c1000_ss_devboard
-  peripheral.adc.resolution_8:
-    extra_configs:
-      - CONFIG_ADC_QMSI_SAMPLE_WIDTH=8
-    platform_whitelist: quark_se_c1000_ss_devboard
+    filter: not CONFIG_SOC_QUARK_D2000
```
Comment (MaureenHelm) [inline tests/drivers/adc/adc_api/testcase.yaml]: Instead of a filter, remove the adc line from `boards/x86/quark_d2000_crb/quark_d2000_crb.yaml`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
This is much better, thank you. Just a couple of minor issues left.

File: samples/sensor/grove_temperature/src/main.c (hunk 9 above)
Comment (nashif) [reply samples/sensor/grove_temperature/src/main.c]: please provide a meaningful message here, SAMPLE seems like a leftover debug message.

Comment (punitvara) [reply samples/sensor/grove_temperature/src/main.c]: Done.

File: tests/drivers/adc/adc_api/testcase.yaml (hunk 10 above)
Comment (punitvara) [reply tests/drivers/adc/adc_api/testcase.yaml]: Done

**Comment by punitvara** [comment]
@nashif Done. Please let me know if anything need to be addressed.

File: samples/sensor/grove_temperature/prj.conf [hunk 11]
```diff
@@ -7,4 +7,3 @@ CONFIG_GROVE=y
 CONFIG_GROVE_TEMPERATURE_SENSOR_ADC_CHANNEL=10
 CONFIG_GROVE_TEMPERATURE_SENSOR=y
 CONFIG_GROVE_TEMPERATURE_SENSOR_V1_X=y
-CONFIG_GROVE_LCD_RGB=y
```
Comment (nashif) [inline samples/sensor/grove_temperature/prj.conf]: why is this being removed? isn't that part of the sample to display the sensor data on the LCD?

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/sensor/grove_temperature/prj.conf (hunk 11 above)
Comment (nashif) [reply samples/sensor/grove_temperature/prj.conf]: On some boards you will need to use 2 pull-up resistors (10k Ohm) between the
SCL/SDA lines and 3.3V

Comment (punitvara) [reply samples/sensor/grove_temperature/prj.conf]: @nashif After connecting two resistors of 10k connecting to SCA and SDA lines , I can see output temperature value on LCD. I am going to push it now.

**Review by nashif — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Looks good. I pulled your commits into #7691

**Pull request closed without merging**